plugins/image-prioritizer/class-image-prioritizer-background-image-styled-tag-visitor.php

```diff
@@ -71,6 +95,55 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 			);
 		}
 
+		$this->lazy_load_bg_images( $context );
+
 		return true;
 	}
+
+	/**
+	 * Optimizes an element with a background image based on whether it is displayed in any initial viewport.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context, with the cursor currently at block with a background image.
+	 */
+	private function lazy_load_bg_images( OD_Tag_Visitor_Context $context ): void {
+		$processor = $context->processor;
+
+		// Lazy-loading can only be done once there are URL Metrics collected for both mobile and desktop.
+		if (
+			$context->url_metric_group_collection->get_first_group()->count() === 0
+			||
+			$context->url_metric_group_collection->get_last_group()->count() === 0
+		) {
+			return;
+		}
+
+		$xpath = $processor->get_xpath();
+
+		// If the element is in the initial viewport, do not lazy load its background image.
+		if ( false !== $context->url_metric_group_collection->is_element_positioned_in_any_initial_viewport( $xpath ) ) {
+			return;
+		}
+
+		$processor->add_class( self::LAZY_BG_IMAGE_CLASS_NAME );
+
+		if ( ! $this->added_lazy_styles ) {
+			$processor->append_head_html(
+				'<style>
```

Comment:
Maybe this should be pulled in from a CSS file like we're doing for the lazy script? In this way we can use the minified version when not in `SCRIPT_DEBUG` mode.

Reply:
Currently, the Webpack setup doesn't minify CSS files the same way it handles JavaScript, as the TerserPlugin (used by CopyWebpackPlugin) is specifically for JavaScript minification. I'll explore other approaches to address this.

Reply:
I guess `wp-scripts` should have some way of doing this? I don't have a lot of experience with setting this up.

Reply:
While `wp-scripts` does handle CSS minification as part of its default build pipeline when importing CSS in JavaScript, it doesn't offer a direct solution for standalone CSS files like these, especially when the goal is to output them in the same directory as the source files.

I think it would be better to use a minifier like `cssnano`—which is also used by the `postcss-loader` in the Webpack configuration provided by `wp-scripts`—through a Webpack plugin like `CssMinimizerPlugin`, with identical `minimizerOptions` as in `wp-scripts` to ensure consistency with the `wp-scripts` build process.

Reply:
Done in a21dec4.